mmseg/datasets/basesegdataset.py

```diff
@@ -235,7 +235,9 @@ def load_data_list(self) -> List[dict]:
         data_list = []
         img_dir = self.data_prefix.get('img_path', None)
         ann_dir = self.data_prefix.get('seg_map_path', None)
-        if osp.isfile(self.ann_file):
+        if not osp.isdir(self.ann_file) and self.ann_file:
```

Comment:
seems not right when `ann_file` is a `dir`

Reply:
If I'm understanding correctly, `ann_file` is not a directory, but some `txt` file for mmsegmentation.
If we want to load from some directory, we would use `ann_dir` and `img_dir` right?

Reply:
yes, by default `ann_file` should be a file or '', but I think we need to take `dir` into consideration, just in case.

Reply:
The old code also ignores the case when `ann_file` is a directory.
I'm not quite sure what you want to do here... Since `ann_file` already has the name `file`, it is assumed that the it is a annotation file rather than a directory. When `ann_file` is `''`, it will automatically turn into a directory anyway since mmengine joins `ann_file` with `data_root`.

Reply:
actually the origin version would lead to `else` if `ann_file` is not a file, what I would like to recommend to do was just to make sure all the cases would work correctly since we can not guarantee all the cases are used fully on the default manner.

Reply:
Yes, that is the same as in the revised code. If `ann_file` is not a file, it would lead to `else` as well. Except when `ann_file` is a file, it will raise an error if the file exists or not instead of silently failing and going to `else`.
I had a problem where I have multiple servers and sometimes I forget to make a specific `ann_file` and it will silently load all of the directory instead. I think failing is much better than silently going to `else` in the case where `ann_file` is given.

Reply:
it seems that it would go on top but not else if it's a dir?

Reply:
yes I understand your point, just need to get it right when it's a dir

Reply:
The `not osp.isdir(self.ann_file)` should check this and would go to the `else`, right?

Reply:
oops, so sorry, I made a mistake

Reply:
@csatsurnh  No problem! Thanks for reviewing this.